docs/fastdds/ros2/Discovery_service/ros2_discovery_service.rst

```diff
@@ -0,0 +1,337 @@
+This allows the server to retake the last state it saved in case of a shutdown.
+
+.. image:: ds_backup_example.png
+    :align: center
+
+In different terminals, run the next code to establish a communication over a backup server.
+
+.. code-block:: console
+
+    fastdds discovery -i 0 -l 127.0.0.1 -p 11811 -b
+
+.. code-block:: console
+
+    export ROS_DISCOVERY_SERVER="127.0.0.1:11811"
+    ros2 run demo_nodes_cpp talker --ros-args --remap __node:=talker
+
+.. code-block:: console
+
+    export ROS_DISCOVERY_SERVER="127.0.0.1:11811"
+    ros2 run demo_nodes_cpp listener --ros-args --remap __node:=listener
+
+We will see how a ``SQLite`` database is created in the path we have run the server.
+
+
+
+Discovery partitions
+^^^^^^^^^^^^^^^^^^^^
+
+The **Discovery Service** communication could be used with different servers to split in virtual partitions the
+discovery info.
+This means that two endpoints only would know each other if there is a server or a server network between them.
+
+We are going to execute an example with two different servers. The following image shows an
+schema of the architecture desired:
+
+.. image:: ds_partition_example.png
+    :align: center
+
+With this schema, we are going to see how *Listener 1* is connected to *Talker 1* and *Talker 2*, but *Listener 2* will
+only listen to *Talker 1*.
+
+We run the first server listening in localhost in default port 11811.
+
+.. code-block:: console
+
+    fastdds discovery -i 0 -l 127.0.0.1 -p 11811
+
+In another terminal we run the second server listening in localhost in port another port, in this case 11888.
+
+.. code-block:: console
+
+    fastdds discovery -i 1 -l 127.0.0.1 -p 11888
+
+Now we run each node in a different terminal. Using the *environment variable* ``ROS_DISCOVERY_SERVER`` we decide which
+server they are connected to. Be aware that the `ids must match
+<https://fast-dds.docs.eprosima.com/en/latest/fastdds/env_vars/env_vars.html>`__.
+
+.. code-block:: console
+
+    export ROS_DISCOVERY_SERVER="127.0.0.1:11811;127.0.0.1:11888"
+    ros2 run demo_nodes_cpp talker --ros-args --remap __node:=talker_1
+
+.. code-block:: console
+
+    export ROS_DISCOVERY_SERVER="127.0.0.1:11811;127.0.0.1:11888"
+    ros2 run demo_nodes_cpp listener --ros-args --remap __node:=listener_1
+
+.. code-block:: console
+
+    export ROS_DISCOVERY_SERVER="127.0.0.1:11811"
+    ros2 run demo_nodes_cpp talker --ros-args --remap __node:=talker_2
+
+.. code-block:: console
+
+    export ROS_DISCOVERY_SERVER=";127.0.0.1:11888"
+    ros2 run demo_nodes_cpp listener --ros-args --remap __node:=listener_2
+
+We should see how *Listener 1* is receiving double messages while *Listener 2* is in a different partition of *Talker 2*
+and so it does not listen him.
+
+.. note::
+
+    Once two endpoints know each other, they do not need the server network between them to
+    listen to each other messages.
+
+
+
+Compare Discovery Service with Simple Discovery
```

Comment:
I think this is a great idea, but I'd re-work the text to it looks better on the browser